app/rest/proxy/avatar_store.go

```diff
@@ -85,3 +109,45 @@ func (fs *FSAvatarStore) location(id string) string {
 	partition := checksum64 % 100
 	return path.Join(fs.storePath, fmt.Sprintf("%02d", partition))
 }
+
+// Resizes an image of supported format (PNG, JPG, GIF) to the size of "limit" px of the biggest side
+// (width or height) preserving aspect ratio.
+// Reruns nil if resizing is not needed.
+func resize(img []byte, limit int) ([]byte, error) {
+	if limit <= 0 {
+		return nil, errors.New("limit should be greater than 0")
+	}
+	src, _, err := image.Decode(bytes.NewReader(img))
+	if err != nil {
+		return nil, errors.Wrap(err, "can't decode avatar image")
+	}
+
+	bounds := src.Bounds()
+	w, h := bounds.Dx(), bounds.Dy()
+	if w <= limit && h <= limit || w <= 0 || h <= 0 {
+		// If resizing image is smaller that the limit or has 0 size, return reader unchenged.
+		return nil, nil
```

Comment:
any good reason why we need a special case here? I mean it can just return input as-is